src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1695,6 +1698,21 @@ public function isXmlHttpRequest()
         return 'XMLHttpRequest' == $this->headers->get('X-Requested-With');
     }
 
+    /**
+     * Adds a Link HTTP header to the corresponding response, using the WebLink component.
+     *
+     * @see https://tools.ietf.org/html/rfc5988
+     */
+    public function addLink(Link $link)
+    {
+        if (!class_exists(HttpHeaderSerializer::class)) {
```

Comment:
actually just having `fig/link-util` is enough for this method to work, so we can test for `GenericLinkProvider`.. but i like the suggestion for `symfony/web-link` in the message :))